cmake/CeleritasAddTest.cmake

```
@@ -407,10 +407,16 @@ function(celeritas_add_test SOURCE_FILE)
       # Launch with MPI directly
       _celeritasaddtest_mpi_cmd(_test_cmd "${_np}" "${_EXE_NAME}")
 
-      set(_test_args "${_EXE_ARGS}")
+      set(_test_args)
       if(_filter)
         list(APPEND _test_args "--gtest_filter=${_filter}")
       endif()
+      if(CELERITAS_TEST_XML)
+        string(REGEX REPLACE "[^a-zA-Z0-9_]+" "_" _xml_name "${_TEST_NAME}")
```

```suggestion
        string(REGEX REPLACE "\*" "_all_" _xml_name "${_TEST_NAME}")
        string(REGEX REPLACE "\+" "_pl_" _xml_name "${_xml_name}")
        string(REGEX REPLACE "-" "_exc_" _xml_name "${_xml_name}")
        string(REGEX REPLACE "[^a-zA-Z0-9_]+" "_" _xml_name "${_xml_name}")
```
but on the other hand the gtest test selection is indeed probably rigid enough that the code as-is could be fine.

---

Maybe easier is to allow those characters since they're legal filenames...

---

Legal but 'hard' to use.

---

It only gets globbed up and parsed by the script 🤷‍♂️